tools/cloner-check/main.go

```diff
@@ -0,0 +1,254 @@
+// Command cloner-check is used to verify that each item stored in the
+// cached_mysql in-memory cache properly implements the fleet.Cloner interface.
+//
+// There are two ways to use this command, the first one is typically used by
+// CI and checks that the cacheable items don't have any changes when compared
+// to the current generated files. That is, if a field is added or modified in
+// a cacheable struct, and those changes haven't been reflected in the
+// generated files yet, it will raise an error, ensuring that the developer
+// takes those changes into account in the custom Clone implementation.
+//
+// The --check flag runs this check mode scenario, but it is optional as running
+// in check mode is the default:
+//
+//	$ go run ./tools/cloner-check/main.go [--check]
+//
+// The second way to use this command is with the --update flag, which is used
+// to update the generated files with the current version of the cacheable
+// items (i.e. the current struct definition). Use this when you've
+// double-checked that the custom Clone implementation is up-to-date and
+// correct.
+//
+//	$ go run ./tools/cloner-check/main.go --update
```

Comment:
spitballing: should we add `make` targets for this?

Reply:
Maybe? I didn't want to make it part of the existing `generate-xxx` because I wanted that to be a very deliberate decision to generate those. I guess for consistency it'd be good to have it in the `Makefile` too, I'll add it.

Reply:
Added.